server/middlewares/ArticleValidation.js

```diff
@@ -105,15 +105,50 @@ class ArticleValidation {
   }
 
   /**
-   * Validate the Article input field
+   * This method validates the tag query
    * @param {object} req - The request object
-   * @param {object} res - The response object
-   * @param {object} next - The callback function to the next middleware.
-   * @return {void}
+   * @returns {void}
    */
-  static validateArticleCommentInput(req, res, next) {
-    ArticleValidation.validateBody(req);
-    sendFormattedError(req, res, next);
+  static validateTagQuery(req) {
+    if (req.query.tag) {
+      // custom validation to check if tags are an strings
+      req.checkQuery(
+        'tag',
+        'tag query must be an integer'
```

Comment:
Why must tat be an integer?

Reply:
@maranatha on the frontend we will populate a drop down with already existing tags. So when a user types in the tag filter box he will select an already existing tag, which will be converted to an integer in the query. This way people can not filter by tags that don't exist

Reply:
This also saves us from having to convert tag names to id's first before working with it